reach

```diff
@@ -115,11 +116,134 @@ run_s () {
+    PACKAGE_MANAGER="apt"
+  fi
+
+  if ( command -v yum > /dev/null 2>&1 ); then
+    PACKAGE_MANAGER="yum"
+  fi
+
+  sudo ${PACKAGE_MANAGER} install "${1}"
+}
+
+timeout_validate () {
+  if [ "$2" -eq "$1" ]; then
+    printf "%s" "$3"
     exit 1
   fi
+}
+
+# shellcheck disable=SC2120
+install_docker () {
+  OPWD=$(pwd)
+  cd / # Depending on `$PWD`, Windows sometimes struggles with paths
+  if (command -v wsl.exe > /dev/null 2>&1); then
+    if (! command -v winget.exe > /dev/null 2>&1); then
+      printf "Installing Winget Windows Package Manager..."
+      WINGET_VERSION="v1.3.431"
+      WINGET_ARTIFACT_NAME="Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle"
+      WINGET_BUNDLE_URL="https://github.com/microsoft/winget-cli/releases/download/${WINGET_VERSION}/${WINGET_ARTIFACT_NAME}"
+      TEMP_DIR="/mnt/c/tmp"
+      WIN_TEMP_DIR="c:\\tmp"
+      DOCKER_BIN="/mnt/c/Program Files/Docker/Docker/resources/bin/docker"
+      mkdir -p $TEMP_DIR
+      sudo wget $WINGET_BUNDLE_URL -O "${TEMP_DIR}/${WINGET_ARTIFACT_NAME}"; powershell.exe "cd ${WIN_TEMP_DIR}; powershell .\\${WINGET_ARTIFACT_NAME}"
+      STOPPER=0
+      # shellcheck disable=SC2039
+      while ( ! command -v winget.exe > /dev/null 2>&1 ); do
+        sleep 1;
+        STOPPER=$((STOPPER + 1))
+        timeout_validate 180 "${STOPPER}" "Winget wasn't installed successfully and timed out!\n"
+      done
+      rm -rf "${TEMP_DIR:?}/${WINGET_ARTIFACT_NAME}"
+    fi
+    printf "\nInstalling Docker through Winget..."
+    powershell.exe 'Start-Process powershell.exe "winget install docker.dockerdesktop --accept-package-agreements --accept-source-agreements" -Verb runAs'
+    STOPPER=0
+    while [ ! -f "${DOCKER_BIN}" ]; do
+      sleep 120;
+      STOPPER=$((STOPPER + 1))
+      timeout_validate 6 "${STOPPER}" "Docker wasn't installed successfully and timed out!\n"
+    done
+    printf "\nThis installation requires a restart for completion; do you want to restart now? [Y,n]: "
+    read -r RESPONSE
+    if [ "${RESPONSE}" = "Y" ]; then
+      call_home "${START_TIME}" "none" "Docker Installed by Reach" "0.0.0" "false" "none" "${1}"
+      powershell.exe 'shutdown -r -t 0'
+    fi
+  else
+    printf "Make sure you have Docker installed following these steps:\n"
+    printf "Ubuntu: https://docs.docker.com/engine/install/ubuntu/\n"
+    printf "Debian: https://docs.docker.com/engine/install/debian/\n"
+    printf "CentOS: https://docs.docker.com/engine/install/centos/\n"
+    printf "Fedora: https://docs.docker.com/engine/install/fedora/\n"
+    printf "RedHat: https://docs.docker.com/engine/install/rhel/\n"
+    printf "  SLES: https://docs.docker.com/engine/install/SLES/\n"
+  fi
+  # shellcheck disable=SC2164
+  cd "${OPWD}"
+}
+
+gen_reachc_id
+
+for DEP in make curl docker docker-compose; do
+  if ! (command -v "${DEP}" >/dev/null 2>&1); then
+    call_home "${START_TIME}" "none" "${DEP} is not installed" "0.0.0" "false" "none" "${1}"
```

Comment:
```suggestion
    call_home "$DEP is not installed" "0.0.0" "none" "false" "script_deps"
```

I don't understand what your previous arguments were trying to do but to me it looks like they weren't right... Did you query some records these produced in Dynamo?